tasks/human_eval_013.rs

```diff
@@ -7,9 +7,194 @@ HumanEval/13
 */
 use vstd::prelude::*;
 
+// TODO: Put your solution (the specification, implementation, and proof) to the task here
 verus! {
 
-// TODO: Put your solution (the specification, implementation, and proof) to the task here
+// This function is part of the specification
+pub open spec fn mul(a: nat, b: nat) -> nat {
+    builtin::mul(a, b)
+}
+
+// This function is also part of the specification
+pub open spec fn divides(factor: nat, candidate: nat) -> bool {
+    exists|k: nat| mul(factor, k) == candidate
+}
+
+// This function is also part of the specification
+spec fn gcd(a: nat, b: nat) -> nat
+    decreases a, b,
+{
+    if a == 0 {
+        b
+    } else if b == 0 {
+        a
+    } else if a > b {
+        gcd(b, a % b)
+    } else {
+        gcd(a, b % a)
+    }
+}
+
+fn compute_gcd(a: u64, b: u64) -> (g: u64)
+    ensures
+        g == gcd(a as nat, b as nat),
+{
+    let mut a1 = a;
+    let mut b1 = b;
+    loop
+        invariant
+            gcd(a1 as nat, b1 as nat) == gcd(a as nat, b as nat),
+    {
+        if a1 == 0 {
+            return b1;
+        } else if b1 == 0 {
+            return a1;
+        } else if a1 > b1 {
+            let m = a1 % b1;
+            a1 = b1;
+            b1 = m;
+        } else {
+            b1 = b1 % a1;
+        }
+    }
+}
+
+// Helper function to prove gcd is a divisor of a and b
```

Comment:
If this is a helper function, I would expect it to be used somewhere.  For this and the lemmas below, it seems like your goal is to go beyond a standard proof and add some assurance that your spec is correct.  That's fine, but I think communicating that via one top-level lemma with appropriate comments would be helpful.